soc/xtensa/intel_s1000/soc.h

```diff
@@ -208,7 +208,7 @@ struct soc_global_regs {
 #define SOC_DCACHE_FLUSH(addr, size)		\
 	xthal_dcache_region_writeback((addr), (size))
 #define SOC_DCACHE_INVALIDATE(addr, size)	\
-	xthal_dcache_region_invalidate((addr), (size))
+	xthal_dcache_region_writeback_inv((addr), (size))
```

Comment:
This seems mixed up.  The macro is still called "invalidate", and now it's doing a "flush then invalidate".  That's going to be confusing at best, and wrong for some uses (consider a user-mapped DMA buffer being filled by the other side of the cache, you don't want to flush whatever garbage the user might have written!).

If the user of this code has writes active in memory and doesn't actually mean "please invalidate the cache", the bug is probably with the caller and not the implementation of "invalidate".